# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/base/ViewportUtils.h
# Commit: 204881474cc1
# Full Hash: 204881474cc1f9d734ba0e14be54f9207c78e8cd
# Author: Botond Ballo <botond@mozilla.com>
# Date: 2020-04-28 16:30:11
# Regressor Bug: 1556556
# File Overlap Count: 1
# Description:
#   Bug 1556556 - Add ViewportUtils::IsZoomedContentRoot(). r=kats,mattwoodrow
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D68727
# ==============================================================================

diff -r 387320881876 -r 204881474cc1 layout/base/ViewportUtils.h
--- a/layout/base/ViewportUtils.h	Tue Apr 28 01:36:24 2020 +0000
+++ b/layout/base/ViewportUtils.h	Tue Apr 28 01:36:26 2020 +0000
@@ -50,6 +50,17 @@
   static nsPoint VisualToLayout(const nsPoint& aPt, PresShell* aContext);
   static nsRect VisualToLayout(const nsRect& aRect, PresShell* aContext);
   static nsPoint LayoutToVisual(const nsPoint& aPt, PresShell* aContext);
+
+  /**
+   * Returns non-null if |aFrame| is inside the async zoom container but its
+   * parent frame is not, thereby making |aFrame| a root of a subtree of
+   * frames representing content that is zoomed in. The value returned in such
+   * cases is the root scroll frame inside the async zoom container.
+
+   * Callers use this to identify points during frame tree traversal where the
+   * visual-to-layout transform needs to be applied.
+   */
+  static const nsIFrame* IsZoomedContentRoot(const nsIFrame* aFrame);
 };
 
 // Forward declare explicit instantiations of GetVisualToLayoutTransform() for
